packages/rln/src/contract/rln_base_contract.ts

```diff
@@ -656,13 +658,18 @@ export class RLNBaseContract {
    * Validates that the rate limit is within the allowed range
    * @throws Error if the rate limit is outside the allowed range
    */
-  private validateRateLimit(rateLimit: number): void {
-    if (
-      rateLimit < RATE_LIMIT_PARAMS.MIN_RATE ||
-      rateLimit > RATE_LIMIT_PARAMS.MAX_RATE
-    ) {
+  private async validateRateLimit(rateLimit: number): Promise<void> {
+    const [minRate, maxRate] = await Promise.all([
+      this.contract.minMembershipRateLimit(),
```

Comment:
this is something that doesn't change during lifecycle of the contract once it is deployed, I believe

if I am not wrong, then you should fetch this value when contract instance gets instantiated

for that I would recommend using a static method similarly how it is done here 
https://github.com/waku-org/js-rln/blob/f6d5deb8cb51602b2b544de5de5b88de96e11aef/src/contract/rln_contract.ts#L56

after that this method becomes sync again

Reply:
Done here: 10aadbe2c805cd7f77124c411dd367e01fbe4995